tests/conditional_formatting/conditional_formatting_plugin.test.ts

```diff
@@ -679,6 +680,199 @@ describe("conditional format", () => {
     });
   });
 
+  describe("Grid Manipulation Formulas", () => {
+    const formulas = [
+      ["=A1", "=A5"],
+      ["=F1", "=F5"],
+      ["=A5", "=F5"],
+      ["=$F$5", "=$F$10"],
+      ["=F5*5", "=F5*10"],
+      ["=SUM(A1:F5)", "=SUM(A1:F5)"],
+    ];
+
+    function cellIsRuleEqual(f: string, useless: string): ConditionalFormatRule {
+      return {
+        type: "CellIsRule",
+        values: [f],
+        operator: "Equal",
+        style: { fillColor: "orange" },
+      };
+    }
+
+    function cellIsRuleBetween(f: string, f2: string): ConditionalFormatRule {
+      return {
+        type: "CellIsRule",
+        values: [f, f2],
+        operator: "Between",
+        style: { fillColor: "orange" },
+      };
+    }
+
+    function colorScaleRule(min: string, max: string): ConditionalFormatRule {
+      return {
+        type: "ColorScaleRule",
+        minimum: {
+          type: "formula",
+          value: min,
+          color: 0,
+        },
+        maximum: {
+          type: "formula",
+          value: max,
+          color: 0,
+        },
+      };
+    }
+
+    function IconSetRule(up: string, lw: string): ConditionalFormatRule {
+      return {
+        type: "IconSetRule",
+        icons: {
+          upper: "u",
+          middle: "m",
+          lower: "l",
+        },
+        upperInflectionPoint: {
+          type: "formula",
+          operator: "gt",
+          value: up,
+        },
+        lowerInflectionPoint: {
+          type: "formula",
+          operator: "gt",
+          value: lw,
+        },
+      };
+    }
+
+    function ruleToCF(rule: ConditionalFormatRule): ConditionalFormat {
+      return {
+        id: "1",
+        rule,
+        ranges: ["A1"],
+      };
+    }
+
+    function formulasToCFs(formulas: string[][], cfrule: any): ConditionalFormat[] {
+      return formulas.map((fn) => cfrule(fn[0], fn[1])).map(ruleToCF);
+    }
+
+    const CFs = [
```

Comment:
```suggestion
    const cfGenerators = [
```
nitpick. Or something similar